source/grid/grid_tools_cache.cc

```diff
@@ -199,6 +199,25 @@ namespace GridTools
     return covering_rtree[level];
   }
 
+  template <int dim, int spacedim>
+  const std::vector<std::set<unsigned int>> &
+  Cache<dim, spacedim>::get_vertex_to_neighbor_subdomain() const
+  {
+    if (update_flags & update_vertex_to_neighbor_subdomain)
```

Comment:
I must admit I don't get the logic. Isn't at the end of the function a `update_flags    = update_flags & ~update_vertex_to_neighbor_subdomain;` missing? Else you will clear and update the data structure at every call?

Reply:
Dear Peter,
To be honest, I am really unsure of how those flags work. I just copied their behavior from another place in the file
e.g. : 
`    if (update_flags & update_vertex_to_cell_map)
`

Reply:
It also contains the cancellation of the bit:

https://github.com/dealii/dealii/blob/e8b1d75b5b26cd125bc7f3b840f28e188a3a0537/source/grid/grid_tools_cache.cc#L63-L68

Reply:
You are right, the bit is never cancelled. I will fix that and push again :)!

Reply:
Now I have a follow-up question: When is this flag actually set, so when do we make the fresh start here?

Reply:
That's a very good questions. I can't seem to find where the initial value of the flag is set. I'll dig deeper. I rebased the PR and test particles/particle_handler_06.mpirun=2.debug is failing, so clearly there is something missing in this PR. I think the triangulation has to specify the update of that flag and right now I did not code that.

Reply:
@kronbichler I change the update_flag to 0xFFF so that the new flag is updated. After discussion with @peterrum  we thought it would be better to just update everything and provision for additional flag in the future by setting update_flag to 0XFFF. The latest push should pass all tests (it passes them on my machine). :). Sorry for the confusion.